src/main/java/com/sweetwith/dailynote/service/FileService.java

```diff
@@ -0,0 +1,145 @@
+import org.springframework.stereotype.Service;
+import org.springframework.web.multipart.MultipartFile;
+import org.springframework.web.multipart.commons.CommonsMultipartFile;
+
+import javax.transaction.Transactional;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+public class FileService
+{
+    private Map<Long, Long> postIdToFileIdMap;
+    private Map<Long, Long> userIdToFileIdMap;
+
+    private FileRepository fileRepository;
+    private Long fileId;
+
+    public FileService(FileRepository fileRepository)
+    {
+        this.fileRepository = fileRepository;
+
+        postIdToFileIdMap = new HashMap<>();
+        userIdToFileIdMap = new HashMap<>();
+
+        fileId = 0L;
+    }
+
+    public void clear()
+    {
+        fileId = 0L;
+    }
+
+    // CREATE
+    @Transactional
+    public FileDto createFile(FileDto fileDto)
+    {
+        java.io.File dir = new java.io.File(fileDto.getFilePath());
+        if (!dir.exists())
+            dir.mkdirs();
+
+        try
+        {
+            fileDto.getMultipartFile().transferTo(new java.io.File(fileDto.getFilePath() + "/" + fileDto.getFileName()));
+        }
+        catch (IOException e)
+        {
+            e.printStackTrace();
+        }
+
+        fileDto.setFileId(++fileId);
+        fileRepository.save(fileDto.toEntity());
+
+        postIdToFileIdMap.put(fileDto.getPostId(), fileDto.getFileId());
+        userIdToFileIdMap.put(fileDto.getUserId(), fileDto.getFileId());
+
+        return fileDto;
+    }
+
+    // READ
+    @Transactional
+    public FileDto readFileByFileId(Long fileId)
+    {
+        File file = fileRepository.findById(fileId).get();
+
+        MultipartFile multipartFile = null;
+        try
+        {
+            java.io.File ioFile = new java.io.File(file.getFilePath() + "/" + file.getFileName());
+            FileItem fileItem = new DiskFileItem(ioFile.getName(), Files.probeContentType(ioFile.toPath()), false, ioFile.getName(), (int) ioFile.length(), ioFile.getParentFile());
+            IOUtils.copy(new FileInputStream(ioFile), fileItem.getOutputStream());
+            multipartFile = new CommonsMultipartFile((org.apache.commons.fileupload.FileItem) fileItem);
+        }
+        catch (Exception e)
+        {
+            e.printStackTrace();
+        }
+
+        FileDto fileDto = FileDto.builder()
+                .fileId(file.getFileId())
+                .fileName(file.getFileName())
+                .filePath(file.getFilePath())
+                .fileType(file.getFileType())
+                .fileAuthor(file.getFileAuthor())
+                .fileSize(file.getFileSize())
+                .postId(file.getPostId())
+                .userId(file.getUserId())
+                .multipartFile(multipartFile)
+                .build();
+
+        return fileDto;
+    }
+    @Transactional
+    public FileDto readFileByPostId(Long postId)
```

Comment:
In case the post has multiple files, how about passing it to a list?

Reply:
OK. I'll fix to return list.